crates/uv-interpreter/python/get_interpreter_info.py

```diff
@@ -415,7 +415,14 @@ def get_operating_system_and_architecture():
     """
     # https://github.com/pypa/packaging/blob/cc938f984bbbe43c5734b9656c9837ab3a28191f/src/packaging/_musllinux.py#L84
     # Note that this is not `os.name`.
-    [operating_system, version_arch] = sysconfig.get_platform().split("-", 1)
+    platform_info = sysconfig.get_platform().split("-", 1)
+    if len(platform_info) == 1:
+        if platform_info[0] == "win32":
+            operating_system, version_arch = "win", "i386"
+        else:
+            raise ValueError(f"Unsupported Platform: {platform_info[0]}")
```

Comment:
We use `"result": "error"` and handle the error in rust code instead of raising exceptions

Reply:
I know, but this is python code, and this function is just to get platform information. For unsupported platforms, you should also exit here. I've thought of two solutions. One is to just exit and add exception printing, and the other is to raise an exception. I think using an exception here might be appropriate.

Reply:
If the python code raises an exception, the rust caller doesn't know what went wrong: Is the platform not supported? Is the version not supported? Is this `python` not a python? Is it a permission error? In rust we get an exit status, stdout and stderr, we can't even tell if we started an actual python interpreter.

Showing a python stacktrace is verbose and not helpful to the user. A user only needs one line (platform not supported) and does not need to see implementation details of our interpreter checking code. Instead, we return that the script ran successfully but the platform was not supported [here](https://github.com/astral-sh/uv/blob/c180fedbce60d8c58611d18cf351ef904fa2280d/crates/uv-interpreter/python/get_interpreter_info.py#L488-L497) and transform it into a rust error [here](https://github.com/astral-sh/uv/blob/7d285148b26550842ba7bb574fa7187faab183f9/crates/uv-interpreter/src/interpreter.rs#L346-L347). This way we get a clear, concise error message:

```
$ uv venv
  × Can't use Python at `/home/konsti/projects/uv/.venv/bin/python3`
  ╰─▶ Unknown operation system: `wedontknowthisos`
```

Reply:
Ok, thanks for your answer